nix/postgres.nix

```diff
@@ -236,153 +231,159 @@ in
+                    cp ${configFile} "$PGDATA/postgresql.conf"
+                  '';
+                in
+                ''
+                  export PGDATA="${config.dataDir}"
+                  ${lib.getExe setupScript}
+                '';
 
-        # DB process
-        ${cfg.name} =
-          let
-            startScript = pkgs.writeShellApplication {
-              name = "start-postgres";
-              text = ''
-                set -x
-                export PATH="${postgresPkg}"/bin:$PATH
-                PGDATA=$(readlink -f "${cfg.dataDir}")
-                export PGDATA
-                postgres -k "$PGDATA"
-              '';
+              # DB process
+              ${name} =
+                let
+                  startScript = pkgs.writeShellApplication {
+                    name = "start-postgres";
+                    text = ''
+                      set -x
+                      export PATH="${postgresPkg}"/bin:$PATH
+                      PGDATA=$(readlink -f "${config.dataDir}")
+                      export PGDATA
+                      postgres -k "$PGDATA"
+                    '';
+                  };
+                in
+                {
+                  command = startScript;
+                  depends_on."${name}-init".condition = "process_completed_successfully";
+                  # SIGINT (= 2) for faster shutdown: https://www.postgresql.org/docs/current/server-shutdown.html
+                  shutdown.signal = 2;
+                  readiness_probe = {
+                    exec.command = "${postgresPkg}/bin/pg_isready -h $(readlink -f ${config.dataDir}) -d template1";
+                    initial_delay_seconds = 2;
+                    period_seconds = 10;
+                    timeout_seconds = 4;
+                    success_threshold = 1;
+                    failure_threshold = 5;
+                  };
+                  # https://github.com/F1bonacc1/process-compose#-auto-restart-if-not-healthy
+                  availability.restart = "on_failure";
+                };
             };
-          in
-          {
-            command = startScript;
-            depends_on."${cfg.name}-init".condition = "process_completed_successfully";
-            # SIGINT (= 2) for faster shutdown: https://www.postgresql.org/docs/current/server-shutdown.html
-            shutdown.signal = 2;
-            readiness_probe = {
-              exec.command = "${postgresPkg}/bin/pg_isready -h $(readlink -f ${cfg.dataDir}) -d template1";
-              initial_delay_seconds = 2;
-              period_seconds = 10;
-              timeout_seconds = 4;
-              success_threshold = 1;
-              failure_threshold = 5;
-            };
-            # https://github.com/F1bonacc1/process-compose#-auto-restart-if-not-healthy
-            availability.restart = "on_failure";
-          };
+        };
       };
+    }));
+  };
+  config = let mergeMapAttrs = f: attrs: lib.mkMerge (lib.mapAttrsToList f attrs); in {
```

Comment:
Maybe move `mergeMapAttrs` elsewhere for other services to use it as well? Or would it be a wiser option to just inline this? @srid any thoughts?

Reply:
Maybe latter: https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)